### DISCUSSION TITLE
How to create macro?

### DISCUSSION THREAD

**Discussion by kossa** [spatie/laravel-html#198] [Q&A]
Hi,
I want to create custom element which I'll use Bootstrap/TailwindCss class for form group, I did not find a way to create a macro in the documentation

**Comment by bskl** [comment]
I define the macro with use Spatie\Html\Html class, within the boot method of App\Providers\AppServiceProvider class:

```php
use Money\Money;
use Spatie\Html\Html;

Html::macro('money', function (string $name, string|Money $value = null) {
    /** @var \Spatie\Html\Html $this */
    $value = $this->value($name, $value);

    if ($value instanceof Money) {
        $value = $value->getAmount();
    }

    return $this->text($name, $value);
});
```

and calling in blade files with this

```php
{{ html()->money('amount') }}
```

**Reply by kamleshwebtech** [reply]
Thank you so much @bskl  Your awesome solution saved my life :) :) 💯

**Comment by francoism90** [comment]
Instead of using Macros, you could use a Mixin. It's basically the same as defining a Marco, but you'll will be able to extend a class as well.

Both have advances and disadvantages. I prefer a Mixin when I really need to extend multiple things at once.
